libs/utils/src/tracing_span_assert.rs

```diff
@@ -164,9 +164,7 @@ fn tracing_subscriber_configured() -> bool {
     tracing::dispatcher::get_default(|d| {
         // it is possible that this closure will not be invoked, but the current implementation
         // always invokes it
-        noop_configured = d
-            .downcast_ref::<tracing::subscriber::NoSubscriber>()
-            .is_some();
+        noop_configured = d.is::<tracing::subscriber::NoSubscriber>();
```

Comment:
Is this change functionally relevant for this PR?

Reply:
I could separate it but it's a sort of follow-up to #4682, which was before I understood that the tenant harness sets up the tracing unconditionally :D (I just didn't see any). I will PR a fix that ... later.